packages/sit-onyx/src/components/OnyxResizeHandle/OnyxResizeHandle.vue

```diff
@@ -0,0 +1,149 @@
+<script setup lang="ts">
+import { computed, onMounted, ref, watch } from "vue";
+import { getTemplateRefElement } from "../../composables/useResizeObserver";
+import { injectI18n } from "../../i18n";
+import type { OnyxResizeHandleProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxResizeHandleProps>(), {
+  min: 1,
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the width should be updated while resizing.
+   * Note: You must take care of actually applied the desired CSS to apply the width.
+   */
+  updateWidth: [width: number];
+  /**
+   * Emitted when the size should be set automatically (by double clicking).
+   */
+  autoSize: [];
+  /**
+   * Emitted when the resizing has started (onMousedown).
+   */
+  start: [];
+  /**
+   * Emitted when the resizing has ended (onMouseup).
+   */
+  end: [];
+}>();
+
+const { t } = injectI18n();
+
+const previousWidth = ref<number>();
+const currentElement = computed(() => getTemplateRefElement(props.element));
+let abortController: AbortController | undefined;
+
+const isActive = ref(false);
+
+const handleMousedown = () => {
+  if (!currentElement.value) return;
+
+  previousWidth.value = currentElement.value.getBoundingClientRect().width;
+  isActive.value = true;
+
+  abortController = new AbortController();
+  const options: AddEventListenerOptions = { signal: abortController.signal, passive: true };
+  window.addEventListener("mousemove", onMouseMove, options);
+  window.addEventListener("mouseup", abort, options);
+  window.addEventListener("keydown", onKeydown, options);
+
+  emit("start");
+};
+
+onMounted(() => {
+  watch(
+    () => props.active,
+    (newActive) => {
+      if (newActive == undefined) return;
+      if (newActive && !isActive.value) handleMousedown();
+      else if (!newActive && isActive.value) abort();
+    },
+    { immediate: true },
+  );
+});
+
+const handleDoubleClick = () => {
+  emit("autoSize");
+};
+
+const onMouseMove = (event: MouseEvent) => {
+  if (!currentElement.value || !isActive.value) return;
+  const width = event.clientX - currentElement.value.getBoundingClientRect().left;
+  emit("updateWidth", Math.max(props.min, width));
+};
+
+const abort = () => {
+  abortController?.abort();
+  previousWidth.value = undefined;
+  isActive.value = false;
+  emit("end");
+};
+
+const onKeydown = (event: KeyboardEvent) => {
+  if (event.key !== "Escape") return;
+  if (previousWidth.value) emit("updateWidth", previousWidth.value);
+  abort();
+};
+</script>
+
+<template>
+  <button
+    tabindex="-1"
+    type="button"
+    role="presentation"
+    :class="['onyx-component', 'onyx-resize-handle', isActive ? 'onyx-resize-handle--active' : '']"
+    :aria-label="t('resizeHandle.label')"
+    @mousedown="handleMousedown"
+    @dblclick="handleDoubleClick"
+  ></button>
```

Comment:
`OnyxVisuallyHidden` should be able to work here, too and would be preferable over an `aria-label`:
```suggestion
    @mousedown="handleMousedown"
    @dblclick="handleDoubleClick"
  ><OnyxVisuallyHidden>{{ t('resizeHandle.label') }}</OnyxVisuallyHiddenProps></button>
```